src/pages/dataSetNotificationTemplates/Edit.tsx

```diff
@@ -0,0 +1,217 @@
+import {
+    DefaultFormFooter,
+    DefaultSectionedFormSidebar,
+    SectionedFormLayout,
+} from '../../components'
+import { SectionedFormProvider, DEFAULT_FIELD_FILTERS } from '../../lib'
+import { DataSetNotificationsFormFields } from './form/DataSetNotificationsFormFields'
+import {
+    getInitialValuesFromTemplate,
+    DataSetNotificationTemplate,
+    DataSetNotificationFormValues,
+    transformFormValues,
+} from './form/getInitialValuesFromTemplate'
+import { useOnEditNotifications } from './form/useOnEditNotifications'
+
+type Params = {
+    id: string
+}
+
+const fieldFilters = [
+    ...DEFAULT_FIELD_FILTERS,
+    'name',
+    'code',
+    'subjectTemplate',
+    'messageTemplate',
+    'notificationRecipient',
+    'dataSetNotificationTrigger',
+    'relativeScheduledDays',
+    'sendStrategy',
+    'recipientUserGroup',
+    'dataSets',
+    'deliveryChannels',
+]
+
+const fetchNotificationTemplate = async (
+    engine: any,
+    id: string
+): Promise<DataSetNotificationTemplate> => {
+    const { notificationTemplate } = await engine.query({
+        notificationTemplate: {
+            resource: `dataSetNotificationTemplates/${id}`,
+            params: {
+                fields: fieldFilters.join(','),
+            },
+        },
+    })
+    return notificationTemplate
+}
+
+const fetchDataSetsByIds = async (engine: any, ids: string[]) => {
+    if (!ids.length) {
+        return []
+    }
+    const { dataSets } = await engine.query({
+        dataSets: {
+            resource: 'dataSets',
+            params: {
+                fields: 'id,name,displayName',
+                filter: `id:in:[${ids.join(',')}]`,
+            },
+        },
+    })
+    return dataSets ?? []
+}
+
+export const Component = () => {
+    const { id: templateId } = useParams<Params>()
+    const dataEngine = useDataEngine()
+    const alert = useAlert(
+        ({ message }) => message,
+        (options) => options
+    )
+
+    const {
+        data: template,
+        isLoading: loadingTemplate,
+        isError: errorTemplate,
+        refetch: refetchTemplate,
+    } = useQuery({
+        queryKey: ['notificationTemplate', templateId],
+        queryFn: () =>
+            fetchNotificationTemplate(dataEngine, templateId as string),
+        enabled: !!templateId,
+    })
+
+    const datasetIds = useMemo(
+        () => template?.dataSets?.map((ds) => ds.id) ?? [],
+        [template]
+    )
+
+    const {
+        data: fetchedDataSets,
+        isLoading: loadingDataSets,
+        isError: errorDataSets,
+        refetch: refetchDataSets,
+    } = useQuery({
+        queryKey: ['datasets', datasetIds],
+        queryFn: () => fetchDataSetsByIds(dataEngine, datasetIds),
+        enabled: datasetIds.length > 0,
+    })
+
+    const isLoading = loadingTemplate ?? loadingDataSets
+    const isError = errorTemplate ?? errorDataSets
+    const onSubmit = useOnEditNotifications(templateId as string)
```

Comment:
Could we not use the generic `useOnSubmitEdit`? I see that `useOnEditNotifications` is not using JSON-patch, is that because we're replicating the old app?